packages/integration-tests/suites/replay/sessionExpiry/test.ts

```diff
@@ -0,0 +1,95 @@
+import { expect } from '@playwright/test';
+import type { ReplayEvent } from '@sentry/types';
+
+import { sentryTest } from '../../../utils/fixtures';
+import { envelopeRequestParser } from '../../../utils/helpers';
+import { getExpectedReplayEvent } from '../../../utils/replayEventTemplates';
+import {
+  getFullRecordingSnapshots,
+  getIncrementalRecordingSnapshots,
+  getReplaySnapshot,
+  normalize,
+  shouldSkipReplayTest,
+  waitForReplayRequest,
+} from '../../../utils/replayHelpers';
+
+// Session should expire after 2s - keep in sync with init.js
+const SESSION_TIMEOUT = 2000;
+
+sentryTest('handles an expired session', async ({ getLocalTestPath, page }) => {
+  if (shouldSkipReplayTest()) {
+    sentryTest.skip();
+  }
+
+  const reqPromise0 = waitForReplayRequest(page, 0);
+  const reqPromise1 = waitForReplayRequest(page, 1);
+
+  await page.route('https://dsn.ingest.sentry.io/**/*', route => {
+    return route.fulfill({
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify({ id: 'test-id' }),
+    });
+  });
+
+  const url = await getLocalTestPath({ testDir: __dirname });
+
+  await page.goto(url);
+
+  const replayEvent0 = envelopeRequestParser(await reqPromise0) as ReplayEvent;
+  expect(replayEvent0).toEqual(getExpectedReplayEvent({}));
+
+  const fullSnapshots0 = getFullRecordingSnapshots(await reqPromise0);
+  expect(fullSnapshots0.length).toEqual(1);
+  const stringifiedSnapshot = normalize(fullSnapshots0[0]);
+  expect(stringifiedSnapshot).toMatchSnapshot('snapshot-0.json');
+
+  // We wait for another segment 0
+  const reqPromise2 = waitForReplayRequest(page, 0);
+
+  await page.click('#button1');
+  const replayEvent1 = envelopeRequestParser(await reqPromise1) as ReplayEvent;
+  expect(replayEvent1).toEqual(getExpectedReplayEvent({ replay_start_timestamp: undefined, segment_id: 1, urls: [] }));
+
+  const fullSnapshots1 = getFullRecordingSnapshots(await reqPromise1);
+  expect(fullSnapshots1.length).toEqual(0);
+
+  const incrementalSnapshots1 = getIncrementalRecordingSnapshots(await reqPromise1);
+  // The number of incremental snapshots depends on the browser
+  expect(incrementalSnapshots1.length).toBeGreaterThanOrEqual(4);
+
+  expect(incrementalSnapshots1).toEqual(
+    expect.arrayContaining([
+      {
+        source: 1,
+        positions: [
+          {
+            id: 9,
+            timeOffset: expect.any(Number),
+            x: expect.any(Number),
+            y: expect.any(Number),
+          },
+        ],
+      },
+    ]),
+  );
```

Comment:
I'm not entirely sure what this snapshot element represents - should we add a comment what we're checking here?
Also related but no action required: Any specific reason why we're checking this explicitly over using `toMatchSnapshot`? (not saying this needs to change)

Reply:
the reason was that I didn't want to test the exact x/y values as I figured they may be flaky - and the `toMatchSnapshot` stuff does string comparison, basically 😬 

I'll add a comment - not sure if this actually helps us at all, my thought was to test that _something_ reasonable is sent, but since it is sadly not stable between browsers, it's pretty diluted now 😅

Reply:
Yeah, I also noticed that incremental snapshots are quite browser-specific... In the errors tests, I opted to check for "length of the incr. snapshot > SomeLowerBoundIObservedDuringTestFlakesMinusOne" 😅
 
RE flakes: My "replay across multiple pages" test snapshot-tests incr. snapshots and so far I haven't noticed flakes (🤞). Perhaps it's still worth a shot? Although, I'm only testing on chromium in this test...

Reply:
Jup, I suspect the reason it works well there is that it's chromium only 😅